### PR TITLE
[DOXIA-727] Fix escaping inside code spans

### PR DESCRIPTION
**Pull request by kwin**
Code cleanup to keep track of surrounding element contexts Remove redundant flags
Get rid of deprecated classes

### CODE REVIEW

File: doxia-core/src/test/java/org/apache/maven/doxia/sink/impl/AbstractSinkTest.java [hunk 1]
```diff
@@ -632,7 +632,7 @@ public void testFigureFromUrl() {
         if (isXmlSink()) {
             assertThat(wrapXml(actual), isIdenticalTo(wrapXml(expected)));
         } else {
-            assertEquals(actual, expected);
+            assertEquals(expected, actual);
```
Comment (michael-o) [inline doxia-core/src/test/java/org/apache/maven/doxia/sink/impl/AbstractSinkTest.java:635]: These are quite embrassing...

File: doxia-modules/doxia-module-markdown/src/main/java/org/apache/maven/doxia/module/markdown/MarkdownSink.java [hunk 2]
```diff
@@ -64,56 +66,81 @@ public class MarkdownSink extends AbstractTextSink implements MarkdownMarkup {
...
 
-    /**  listStyles. */
-    private final Stack<String> listStyles;
+    /** Keep track of end markup for inline events. */
+    protected Queue<Queue<String>> inlineStack = Collections.asLifoQueue(new LinkedList<>());
 
-    /** Keep track of the closing tags for inline events. */
-    protected Stack<List<String>> inlineStack = new Stack<>();
+    /** The context of the surrounding elements as stack (LIFO) */
+    protected Queue<ElementContext> elementContextStack = Collections.asLifoQueue(new LinkedList<>());
```
Comment (michael-o) [inline doxia-modules/doxia-module-markdown/src/main/java/org/apache/maven/doxia/module/markdown/MarkdownSink.java:91]: Stupid question: Why not use a stack directly?

**Review by michael-o — APPROVED** [review]
Nice amount of work, no objections here.

**Review by slachiewicz — APPROVED** [review]

File: doxia-modules/doxia-module-markdown/src/main/java/org/apache/maven/doxia/module/markdown/MarkdownSink.java (hunk 2 above)
Comment (kwin) [reply doxia-modules/doxia-module-markdown/src/main/java/org/apache/maven/doxia/module/markdown/MarkdownSink.java:91]: It says in its javadoc
> A more complete and consistent set of LIFO stack operations is provided by the [Deque](https://docs.oracle.com/javase/8/docs/api/java/util/Deque.html) interface and its implementations, which should be used in preference to this class.

Comment (michael-o) [reply doxia-modules/doxia-module-markdown/src/main/java/org/apache/maven/doxia/module/markdown/MarkdownSink.java:91]: I see, thanks!

**Comment by jira-importer** [comment]
Resolve #855

**Pull request merged**